src/nice-path-decomposition.cpp

```diff
@@ -191,4 +191,12 @@ void NicePathDecomposition::enumerate(std::vector <vertex_t> &U) {
             bag.edge.m_target = U[bag.edge.m_target];
         }
     }
+}
+
+void NicePathDecomposition::push_back_bag(Bag b) {
+    _nice_bags.push_back(b);
```

Comment:
Shouldn't we throw an exception in the case the new bag is not consistent with the current decomposition?

Reply:
We can add verification with O(1) time with extra memory O(n) or with O(n) time.